### ISSUE TITLE
Adding a restConfiguration() make all rest properties to be ignored

### ISSUE THREAD

**Issue by nicolaferraro** [apache/camel-k-runtime#266]
A simple file with a rest route:

```
// camel-k: language=java

import org.apache.camel.builder.RouteBuilder;

public class Called extends RouteBuilder {
  @Override
  public void configure() throws Exception {

		//restConfiguration();

		rest().get("/")
		  .route()
		  .setBody().constant("Ciao");
  }
}

```

Running it with `-p camel.rest.port=8080` makes it actually run on port 8080.

But if you remove the comment on `restConfiguration()` it binds itself on port 0.

**Comment by lburgazzoli** [comment]
Requires: https://issues.apache.org/jira/browse/CAMEL-14723

**Comment by lburgazzoli** [comment]
Will be fixed when migrating to Camel 3.2.0

**Comment by lburgazzoli** [comment]
https://github.com/apache/camel-k-runtime/pull/318
